packages/@aws-cdk/aws-eks/lib/k8s-cluster-autoscaler.ts

```diff
@@ -0,0 +1,331 @@
+import autoscaling = require('@aws-cdk/aws-autoscaling');
+import iam = require('@aws-cdk/aws-iam');
+import cdk = require('@aws-cdk/core');
+import { Cluster } from './cluster';
+import { KubernetesResource } from './k8s-resource';
+
+/**
+ * The properties for the Cluster Autoscaler.
+ */
+export interface KubernetesClusterAutoscalerProps {
+
+  /**
+   * The EKS cluster to deploy the cluster autoscaler to.
+   *
+   * [disable-awslint:ref-via-interface]
+   */
+  readonly cluster: Cluster;
+
+  /**
+   * An array of Autoscaling Groups, known as node groups, to configure for autoscaling.
+   *
+   * [disable-awslint:ref-via-interface]
+   */
+  readonly nodeGroups: autoscaling.AutoScalingGroup[];
+
+  /**
+   * The version of the Cluster Autoscaler to deploy.
+   *
+   * @default - the default version is v1.14.6
+   */
+  readonly version?: string;
+
+}
+
+/**
+ * The Cluster Autoscaler Construct. This will create a new IAM Policy, add labels to the ASGs, and
+ * deploy the Cluster Autoscaler manifest.
```

Comment:
Before describing what this does, you should say why users would want to use it.